TRLevelControl/IO/TRLevelReader.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace TRLevelControl;
+
+public class TRLevelReader : BinaryReader
+{
+    public TRLevelReader(Stream stream)
+        : base(stream) { }
+}
```

Comment:
I don't quite understand the purpose of this class, what are our plans for it?

Reply:
It will hold utility methods for reading commonly across all levels. So for example, the following will be in this reader class and TR1-3 can call it directly on the reader instead of duplicating code in each control class.

```
public List<TRTexImage8> ReadImage8s(long numImages)
{
    List<TRTexImage8> images = new((int)numImages);
    for (long i = 0; i < numImages; i++)
    {
        images.Add(new()
        {
            Pixels = ReadBytes(TRConsts.TPageSize)
        });
    }
    return images;
}
```

And the same will go for the writers. As everything currently is stored in arrays in the levels, the plan is to gradually convert to lists and implement these utility methods at the same time. My thought was just to kickstart these classes as a base as part of this PR for ease.

Reply:
I see. I'd remove the BinaryWriter inheritance though. I think. Ideally, have only one public method – write level, read level, and the rest are internals. WDYT?

Reply:
Yes, perhaps changing it to `has_a` once we get everything ported over to it will be better. So the control classes will dictate the flow e.g. read images first, then rooms for TR1; read palette first, then images for TR2, etc - and the separate helper reader will store a local binary reader and deal with the specifics of how to read the data.